test/coverage/CMakeLists.txt

```diff
@@ -0,0 +1,140 @@
+# ============================================================================
+#                  SeqAn - The Library for Sequence Analysis
+# ============================================================================
+#
+# Copyright (c) 2006-2018, Knut Reinert & Freie Universitaet Berlin
+# Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+# Copyright (c) 2012-2017, Lars Bilke (Adopted from CodeCoverage.cmake; BSD-License)
+#     https://github.com/bilke/cmake-modules/blob/master/CodeCoverage.cmake
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+#       its contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+# DAMAGE.
+# ============================================================================
+
+cmake_minimum_required (VERSION 3.2)
+project (seqan3_test_coverage CXX)
+
+include (../seqan3-test.cmake)
+
+if (NOT (${CMAKE_BUILD_TYPE} STREQUAL "Debug"))
+    message(WARNING "Coverage test must be build in debug mode [build type = ${CMAKE_BUILD_TYPE}]")
+endif ()
+
+find_program (LCOV_COMMAND  NAMES lcov lcov.bat lcov.exe lcov.perl)
+find_program (GENHTML_COMMAND NAMES genhtml genhtml.perl genhtml.bat)
+
+if (NOT LCOV_COMMAND)
+    message(FATAL_ERROR "lcov not found! Aborting...")
+endif ()
+
+set (COVERAGE_EXCLUDES
+    "'/usr/*'"
+    "'${SEQAN3_CLONE_DIR}/submodules/*'"
+    "'${SEQAN3_CLONE_DIR}/test/unit/*'"
+    "'${PROJECT_BINARY_DIR}/vendor/*'"
+)
+set_property (GLOBAL PROPERTY COVERAGE_ALL_TESTS "")
```

Comment:
??? Can you add a comment on how to use this property.
Also the naming does not fit the naming conventions for the projects. I would rather use `TEST_COVERAGE_ALL_TARGETS`.

Reply:
Usage is here: `set_property(GLOBAL APPEND PROPERTY COVERAGE_ALL_TESTS ${target})`